File: ios/MenCare/Core/Data/Fake/FakeNotificationRepository.swift
import Foundation

/// Only the signed-in user's notifications: client, staff and owner each see their own.
@MainActor
final class FakeNotificationRepository: NotificationRepository {

    private let store: InMemoryStore

    init(store: InMemoryStore) {
        self.store = store
    }

    func feed() async throws -> NotificationFeed {
        let mine = store.notifications
            .filter { $0.userId == store.currentUser?.id }
            .sorted { $0.at > $1.at }
        return NotificationFeed(notifications: mine, unreadCount: mine.filter { !$0.read }.count)
    }

    func markAllRead() async -> AppResult<Void> {
        guard let userId = store.currentUser?.id else { return .success(()) }
        store.notifications = store.notifications.map {
            var n = $0
            if n.userId == userId { n.read = true }
            return n
        }
        return .success(())
    }
}

File: android/core/data/src/main/kotlin/com/devora/mencare/core/data/fake/FakeNotificationRepository.kt
package com.devora.mencare.core.data.fake

import com.devora.mencare.core.data.repository.NotificationRepository
import com.devora.mencare.core.model.AppNotification
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.combine
import javax.inject.Inject
import javax.inject.Singleton

/** Only the signed-in user's notifications: client, staff and owner each see their own. */
@Singleton
class FakeNotificationRepository @Inject constructor(
    private val store: InMemoryStore,
) : NotificationRepository {

    override val notifications: Flow<List<AppNotification>> =
        combine(store.notifications, store.currentUser) { list, user ->
            list.filter { it.userId == user?.id }.sortedByDescending { it.at }
        }

    override suspend fun markAllRead() {
        val userId = store.currentUser.value?.id ?: return
        store.notifications.value = store.notifications.value.map {
            if (it.userId == userId) it.copy(read = true) else it
        }
    }
}
